datatypes.go

```diff
@@ -115,6 +133,36 @@ type TriggerData struct {
 	Tags       []string `json:"__notifier_trigger_tags"`
 }
 
+func (trigger TriggerData) PopulateDescription(events NotificationEvents) (string, error) {
```

Comment:
Contrary to its name, this method does not populate a trigger's description. It returns a populated description string. You should either rename this to `GetPopulatedDescription` or return a full new instance of a trigger with populated description field, not just string.